### PR TITLE
bug/1144_insert_null_in_ckan_column_empty_value

### LINKED ISSUE #1144: [cygnus-ngsi][NGSICKANSink] Add patch fixing CKAN bug about "", [], {} and null
When an empty attribute value, i.e. `""`, is notified and the related column in the datastore is of type Json, a `500 server error` arises.

This must be fixed by inserting the `NULL` value instead.

To be fixed for upcoming release/1.3.0 and already released 1.2.0

### PR DESCRIPTION
**Pull request by frbattid**
* Fixes #1144 
* 100% tests passed
* (unofficial) e2e tests passed:

Row mode, `"26.5`":
```
Received data ({  "subscriptionId" : "51c0ac9ed714fb3b37d7d5a8",  "originator" : "localhost",  "contextResponses" : [    {      "contextElement" : {        "attributes" : [          {            "name" : "temperature",            "type" : "centigrade",            "value" : "26.5"          }        ],        "type" : "Room",        "isPattern" : "false",        "id" : "Room1"      },      "statusCode" : {        "code" : "200",        "reasonPhrase" : "OK"      }    }  ]})
Persisting data at OrionCKANSink (orgName=city012, pkgName=city012, resName=room1_room, data={"recvTimeTs": "1473227786","recvTime": "2016-09-07T05:56:26.571Z","fiwareServicePath": "/","entityId": "Room1","entityType": "Room","attrName": "temperature","attrType": "centigrade","attrValue": "26.5"})
```
Row mode, `""`:
```
Received data ({  "subscriptionId" : "51c0ac9ed714fb3b37d7d5a8",  "originator" : "localhost",  "contextResponses" : [    {      "contextElement" : {        "attributes" : [          {            "name" : "temperature",            "type" : "centigrade",            "value" : ""          }        ],        "type" : "Room",        "isPattern" : "false",        "id" : "Room1"      },      "statusCode" : {        "code" : "200",        "reasonPhrase" : "OK"      }    }  ]})
Persisting data at OrionCKANSink (orgName=city012, pkgName=city012, resName=room1_room, data={"recvTimeTs": "1473227815","recvTime": "2016-09-07T05:56:55.850Z","fiwareServicePath": "/","entityId": "Room1","entityType": "Room","attrName": "temperature","attrType": "centigrade"})
```
Row mode, `null`:
```
Received data ({  "subscriptionId" : "51c0ac9ed714fb3b37d7d5a8",  "originator" : "localhost",  "contextResponses" : [    {      "contextElement" : {        "attributes" : [          {            "name" : "temperature",            "type" : "centigrade",            "value" : null          }        ],        "type" : "Room",        "isPattern" : "false",        "id" : "Room1"      },      "statusCode" : {        "code" : "200",        "reasonPhrase" : "OK"      }    }  ]})
Persisting data at OrionCKANSink (orgName=city012, pkgName=city012, resName=room1_room, data={"recvTimeTs": "1473227917","recvTime": "2016-09-07T05:58:37.358Z","fiwareServicePath": "/","entityId": "Room1","entityType": "Room","attrName": "temperature","attrType": "centigrade"})
```
Row mode, `{}`:
```
Received data ({  "subscriptionId" : "51c0ac9ed714fb3b37d7d5a8",  "originator" : "localhost",  "contextResponses" : [    {      "contextElement" : {        "attributes" : [          {            "name" : "temperature",            "type" : "centigrade",            "value" : {}          }        ],        "type" : "Room",        "isPattern" : "false",        "id" : "Room1"      },      "statusCode" : {        "code" : "200",        "reasonPhrase" : "OK"      }    }  ]})
Persisting data at OrionCKANSink (orgName=city012, pkgName=city012, resName=room1_room, data={"recvTimeTs": "1473227994","recvTime": "2016-09-07T05:59:54.382Z","fiwareServicePath": "/","entityId": "Room1","entityType": "Room","attrName": "temperature","attrType": "centigrade"})
```
Row mode, `[]`:
```
Received data ({  "subscriptionId" : "51c0ac9ed714fb3b37d7d5a8",  "originator" : "localhost",  "contextResponses" : [    {      "contextElement" : {        "attributes" : [          {            "name" : "temperature",            "type" : "centigrade",            "value" : []          }        ],        "type" : "Room",        "isPattern" : "false",        "id" : "Room1"      },      "statusCode" : {        "code" : "200",        "reasonPhrase" : "OK"      }    }  ]})
Persisting data at OrionCKANSink (orgName=city012, pkgName=city012, resName=room1_room, data={"recvTimeTs": "1473228089","recvTime": "2016-09-07T06:01:29.659Z","fiwareServicePath": "/","entityId": "Room1","entityType": "Room","attrName": "temperature","attrType": "centigrade"})
```

ckan_row_nulls

Column mode, `"26.5"`:
```
Received data ({  "subscriptionId" : "51c0ac9ed714fb3b37d7d5a8",  "originator" : "localhost",  "contextResponses" : [    {      "contextElement" : {        "attributes" : [          {            "name" : "temperature",            "type" : "centigrade",            "value" : "26.5"          }        ],        "type" : "Room",        "isPattern" : "false",        "id" : "Room3"      },      "statusCode" : {        "code" : "200",        "reasonPhrase" : "OK"      }    }  ]})
Persisting data at OrionCKANSink (orgName=city012, pkgName=city012, resName=room3_room, data={"recvTime": "2016-09-07T06:04:58.855Z","fiwareServicePath": "/","entityId": "Room3","entityType": "Room","temperature": "26.5"})
```
Column mode, `""`:
```
Received data ({  "subscriptionId" : "51c0ac9ed714fb3b37d7d5a8",  "originator" : "localhost",  "contextResponses" : [    {      "contextElement" : {        "attributes" : [          {            "name" : "temperature",            "type" : "centigrade",            "value" : ""          }        ],        "type" : "Room",        "isPattern" : "false",        "id" : "Room3"      },      "statusCode" : {        "code" : "200",        "reasonPhrase" : "OK"      }    }  ]})
Persisting data at OrionCKANSink (orgName=city012, pkgName=city012, resName=room3_room, data={"recvTime": "2016-09-07T06:05:44.359Z","fiwareServicePath": "/","entityId": "Room3","entityType": "Room"})
```
Column mode, `null`:
```
Received data ({  "subscriptionId" : "51c0ac9ed714fb3b37d7d5a8",  "originator" : "localhost",  "contextResponses" : [    {      "contextElement" : {        "attributes" : [          {            "name" : "temperature",            "type" : "centigrade",            "value" : null          }        ],        "type" : "Room",        "isPattern" : "false",        "id" : "Room3"      },      "statusCode" : {        "code" : "200",        "reasonPhrase" : "OK"      }    }  ]})
Persisting data at OrionCKANSink (orgName=city012, pkgName=city012, resName=room3_room, data={"recvTime": "2016-09-07T06:06:07.997Z","fiwareServicePath": "/","entityId": "Room3","entityType": "Room"})
```
Column mode, `{}`:
```
Received data ({  "subscriptionId" : "51c0ac9ed714fb3b37d7d5a8",  "originator" : "localhost",  "contextResponses" : [    {      "contextElement" : {        "attributes" : [          {            "name" : "temperature",            "type" : "centigrade",            "value" : {}          }        ],        "type" : "Room",        "isPattern" : "false",        "id" : "Room3"      },      "statusCode" : {        "code" : "200",        "reasonPhrase" : "OK"      }    }  ]})
Persisting data at OrionCKANSink (orgName=city012, pkgName=city012, resName=room3_room, data={"recvTime": "2016-09-07T06:06:34.394Z","fiwareServicePath": "/","entityId": "Room3","entityType": "Room"})
```
Column mode, `[]`:
```
Received data ({  "subscriptionId" : "51c0ac9ed714fb3b37d7d5a8",  "originator" : "localhost",  "contextResponses" : [    {      "contextElement" : {        "attributes" : [          {            "name" : "temperature",            "type" : "centigrade",            "value" : []          }        ],        "type" : "Room",        "isPattern" : "false",        "id" : "Room3"      },      "statusCode" : {        "code" : "200",        "reasonPhrase" : "OK"      }    }  ]})
Persisting data at OrionCKANSink (orgName=city012, pkgName=city012, resName=room3_room, data={"recvTime": "2016-09-07T06:07:10.88Z","fiwareServicePath": "/","entityId": "Room3","entityType": "Room"})
```

ckan_column_nulls

### CODE REVIEW

**Comment by pcoello25** [comment]
LGTM, passing the ball to @fgalan

File: CHANGES_NEXT_RELEASE [hunk 1]
```diff
@@ -11,3 +11,4 @@
 - [cygnus][doc] Add Orion-Cygnus-Kafka integration example (#1132)
 - [cygnus-ngsi][hardening] Decrease to DEBUG the logging level of batching related traces (#1126)
 - [cygnus-common][bug] Release connection upon persistence error (#1139)
+- [cygnus-common][bug] Insert NULL in CKAN columns when an empty ("", null, {} or []) attribute value is notified (#1144)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]: What about metadata? Did you something about it at the end in this PR?

Comment (frbattid) [inline CHANGES_NEXT_RELEASE]: Yes, metadata was tuned as well, I'll change this line to make it clear.

Comment (frbattid) [inline CHANGES_NEXT_RELEASE]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1148/commits/c8e06e24f337c9ee971b6021cfd955f3fd1e7aea

**Comment by fgalan** [comment]
LGTM

**Pull request merged**